### DISCUSSION TITLE
Request for clarification of TTL message functions in documentation

### DISCUSSION THREAD

**Discussion by PhilHannent** [weyoss/redis-smq#118] [Q&A] [answered]
I am looking for clarification of the following function calls, in terms of their behaviour:

https://github.com/weyoss/redis-smq/blob/master/docs/api/classes/ProducibleMessage.md#setttl

https://github.com/weyoss/redis-smq/blob/master/docs/api/classes/ProducibleMessage.md#setconsumetimeout

I have a queue on which I put messages onto at specific intervals and I'd like to ensure the existing queued item is either dropped (or ideally, that I can't add a duplicate message until the queued item has been processed).

Thanks

**Comment by weyoss** [comment, marked as answer]
Messages can be [set to expire](https://github.com/weyoss/redis-smq/blob/master/docs/api/classes/ProducibleMessage.md#setttl) when not delivered within a given amount of time or to have a [consumption timeout](https://github.com/weyoss/redis-smq/blob/master/docs/api/classes/ProducibleMessage.md#setconsumetimeout) while being in process.
